pyramid_tm/__init__.py

```diff
@@ -126,4 +136,18 @@ def includeme(config):
     - If none of the above conditions are True, the transaction will be
       committed (via ``transaction.commit()``).
     """
+    config.add_request_method(
+        'pyramid_tm.create_tm', name='transaction', reify=True,
+    )
     config.add_tween('pyramid_tm.tm_tween_factory', under=EXCVIEW)
+
+    def ensure():
+        manager_hook = config.registry.settings.get("tm.manager_hook")
+        if manager_hook is not None:
+            try:
+                manager_hook = resolver.maybe_resolve(manager_hook)
+                config.registry.settings["tm.manager_hook"] = manager_hook
+            except (ImportError, ValueError):
+                raise ConfigurationError("Unable to resolve tm.manager_hook")
+
+    config.action(None, ensure, order=10)
```

Comment:
I disagree.

What if the user has `pyramid_tm` in their includes, or they call `config.include('pyramid_tm')` and then set the `tm.manager_hook` on the settings?

This is the same reason we deferred loading settings in `pyramid_mako` until `commit()` and not as soon as `includeme()` was called.